test/performance/observed_concurency_test.go

```diff
@@ -165,52 +160,74 @@ func TestObservedConcurrency(t *testing.T) {
 		t.Fatalf("Error creating spoofing client: %v", err)
 	}
 
-	trafficStart := time.Now()
-
-	responseChannel := make(chan *spoof.Response, 1000)
-
-	logger.Infof("Running %d concurrent requests for %v", concurrency, duration)
-	requestsMade, err := generateTraffic(client, url, concurrency, duration, responseChannel)
+	// Make sure we are ready to serve.
+	st := time.Now()
+	logger.Info("Starting to probe the endpoint at ", st)
+	_, err = pkgTest.WaitForEndpointState(
+		clients.KubeClient,
+		logger,
+		domain+"/?timeout=10", // To generate any kind of a valid response.
```

Comment:
I cannot comment on the above, but since you are using domain here you should remove lines 152 to 157 and use domain everywhere instead of endpoint for the url. The spoofing client on line 158 already does a GetServiceEndpoint for a non-resolvable domain.

Reply:
Done